### PR TITLE
Jetpacks

### PR DESCRIPTION
**Pull request by metalgearsloth**
Builds on https://github.com/space-wizards/space-station-14/pull/2893
Branched from https://github.com/space-wizards/space-station-14/pull/8969

1. I shuffled some mover stuff around for the eventual movement refactor
2. TimedDespawn was moved to shared (the client one only despawns clientside entities).

https://user-images.githubusercontent.com/31366439/175057516-a465d680-71fb-4081-b430-406955273f18.mp4

:cl: metalgearsloth & lzimann
- add: Jetpacks have been added.

### CODE REVIEW

File: Resources/Prototypes/Entities/Objects/Tools/jetpacks.yml [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+  components:
+    - type: Sprite
+      sprite: Objects/Tanks/Jetpacks/blue.rsi
+      state: icon
+      netsync: false
+    - type: UserInterface
+      interfaces:
+        - key: enum.SharedGasTankUiKey.Key
+          type: GasTankBoundUserInterface
+    - type: Clothing
```
Comment (Emisse) [inline Resources/Prototypes/Entities/Objects/Tools/jetpacks.yml:16]: can you add a size to this so its heavier than 15 weight. i dont think players should be running around with these in their bags without sacrificing a lot of weight

Comment (metalgearsloth) [reply Resources/Prototypes/Entities/Objects/Tools/jetpacks.yml:16]: Could probably make it like 30 or 50. Generally they get used as backpacks anyway.

Comment (metalgearsloth) [reply Resources/Prototypes/Entities/Objects/Tools/jetpacks.yml:18]: Alright made it 100 so they don't fit in bags normally, except for captain's and mini.

Comment (ZeroDayDaemon) [reply Resources/Prototypes/Entities/Objects/Tools/jetpacks.yml:18]: Captains jetpack should be substantially smaller in size than others as it is intended to fit into a backpack

**Review by DogZeroX — APPROVED** [review]

**Comment by metalgearsloth** [comment]
> looks good, a bit wonky with the rot on the particle effects, maybe get a sprinter to make a circular effect?

I just made them noRot for now.

**Comment by Emisse** [comment]
are you going to add captains jetpack to his locker spawn and also make it a traitor objective in this PR? and will other ones have to be mapped or ordered from cargo

**Comment by metalgearsloth** [comment]
> are you going to add captains jetpack to his locker spawn and also make it a traitor objective in this PR? and will other ones have to be mapped or ordered from cargo

hell no I'm lazy

**Comment by Emisse** [comment]
ok speedmerge and ill do it : )

**Comment by lzimann** [comment]
Thanks for finishing this!

File: Content.Shared/Movement/Components/JetpackComponent.cs [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+using Content.Shared.Atmos;
+using Robust.Shared.GameStates;
+
+namespace Content.Shared.Movement.Components;
+
+[RegisterComponent, NetworkedComponent]
+public sealed class JetpackComponent : Component
+{
+    [ViewVariables(VVAccess.ReadWrite), DataField("enabled")]
+    public bool Enabled = false;
```
Comment (mirrorcult) [inline Content.Shared/Movement/Components/JetpackComponent.cs:11]: does this need to exist when `ActiveJetpack` exists

File: Content.Shared/Movement/Systems/SharedJetpackSystem.cs [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+    {
+        if (!component.Enabled) return;
+        RemComp<JetpackUserComponent>(args.User);
+    }
+
+    private void OnJetpackUnequipped(EntityUid uid, JetpackComponent component, GotUnequippedEvent args)
+    {
+        if (!component.Enabled) return;
+        RemComp<JetpackUserComponent>(args.Equipee);
+    }
```
Comment (mirrorcult) [inline Content.Shared/Movement/Systems/SharedJetpackSystem.cs:57]: shouldnt jetpackuser bookkeeping be handled by the action enabling/disabling. cause its an item action, it already does this stuff

Comment (mirrorcult) [reply Content.Shared/Movement/Systems/SharedJetpackSystem.cs:57]: same with the handequip/equipped stuff below

File: Resources/Prototypes/Entities/Objects/Tools/jetpacks.yml [hunk 4]
```diff
@@ -0,0 +1,293 @@
...
+  suffix: Filled
+  components:
+  - type: GasTank
+    outputPressure: 21.27825
+    air:
+      volume: 70
+      temperature: 293.15
+      moles:
+        - 22.6293856
+
```
Comment (mirrorcult) [inline Resources/Prototypes/Entities/Objects/Tools/jetpacks.yml:236]: remove. improvised jetpacks are unneeded bc of throwing+extinguishers, mining shouldn't get jetpacks

File: Resources/Prototypes/Entities/Objects/Tools/jetpacks.yml [hunk 5]
```diff
@@ -0,0 +1,293 @@
...
+  suffix: Empty
+  components:
+  - type: Sprite
+    sprite: Objects/Tanks/Jetpacks/security.rsi
+  - type: Clothing
+    sprite: Objects/Tanks/Jetpacks/security.rsi
+    Slots:
+      - Back
+
+# Filled mining
```
Comment (mirrorcult) [inline Resources/Prototypes/Entities/Objects/Tools/jetpacks.yml:251]:
```suggestion
# Filled security
```

File: Resources/Textures/Objects/Tanks/Jetpacks/improvised.rsi/meta.json [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+    {
+      "name": "equipped-BACKPACK",
+      "directions": 4
+    },
+    {
+      "name": "on-equipped-BACKPACK",
+      "directions": 4
+    }
+  ]
+}
```
Comment (mirrorcult) [inline Resources/Textures/Objects/Tanks/Jetpacks/improvised.rsi/meta.json:41]: remove etc

**Review by mirrorcult — CHANGES_REQUESTED** [review]

File: Content.Shared/Movement/Systems/SharedJetpackSystem.cs (hunk 3 above)
Comment (metalgearsloth) [reply Content.Shared/Movement/Systems/SharedJetpackSystem.cs:57]: Is there even an event for it? Internals just subscribes to inventory events.

**Pull request merged**